resolver/hosts_file_resolver.go

```diff
@@ -0,0 +1,182 @@
+package resolver
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"strings"
+
+	"github.com/0xERR0R/blocky/model"
+	"github.com/0xERR0R/blocky/util"
+	"github.com/miekg/dns"
+	"github.com/sirupsen/logrus"
+)
+
+const hostsDNSTTL = 3600
+
+type HostsFileResolver struct {
+	NextResolver
+	HostsFilePath string
+	parsed        bool
+	hosts         []host
+}
+
+func (r *HostsFileResolver) handleReverseDNS(request *model.Request) *model.Response {
+	question := request.Req.Question[0]
+	if question.Qtype == dns.TypePTR {
+		response := new(dns.Msg)
+		response.SetReply(request.Req)
+
+		for _, host := range r.hosts {
+			raddr, _ := dns.ReverseAddr(host.IP.String())
+
+			if raddr == question.Name {
+				ptr := new(dns.PTR)
+				ptr.Ptr = dns.Fqdn(host.Hostname)
```

Comment:
I'm not sure: maybe also aliases should be considered here. Example:

`10.0.0.1                router0 router1 router2`

Should reverse DNS for the IP return only router0 as A record or 3 A records (rourter0, router1 and router2)?

Reply:
Ah, I've thought you should only get one PTR record, but DNS does allow for more. That's been fixed.